internal/importers/gvt/importer_test.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2017 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gvt
+
+import (
+	"bytes"
+	"log"
+	"path/filepath"
+	"testing"
+
+	"github.com/golang/dep"
+	"github.com/golang/dep/internal/gps"
+	"github.com/golang/dep/internal/importers/importertest"
+	"github.com/golang/dep/internal/test"
+	"github.com/pkg/errors"
+)
+
+func TestGvtConfig_Convert(t *testing.T) {
+	testCases := map[string]struct {
+		importertest.TestCase
+		gvtConfig gvtManifest
+	}{
+		"package with master branch": {
+			importertest.TestCase{
+				WantConstraint: importertest.V1Constraint,
+				WantRevision:   importertest.V1Rev,
+				WantVersion:    importertest.V1Tag,
+			},
+			gvtManifest{
+				Deps: []gvtPkg{
+					{
+						ImportPath: importertest.Project,
+						Revision:   importertest.V1Rev,
+						Branch:     "master",
+					},
+				},
+			},
+		},
+		"package with non-master branch": {
+			importertest.TestCase{
+				WantConstraint: importertest.V2Branch,
+				WantRevision:   importertest.V2PatchRev,
+				WantVersion:    importertest.V2PatchTag,
+			},
+			gvtManifest{
+				Deps: []gvtPkg{
+					{
+						ImportPath: importertest.Project,
+						Revision:   importertest.V2PatchRev,
+						Branch:     importertest.V2Branch,
+					},
+				},
+			},
+		},
+		"package with HEAD branch": {
+			importertest.TestCase{
+				WantConstraint: "*",
+				WantRevision:   importertest.V1Rev,
+				WantVersion:    importertest.V1Tag,
+			},
+			gvtManifest{
+				Deps: []gvtPkg{
+					{
+						ImportPath: importertest.Project,
+						Revision:   importertest.V1Rev,
+						Branch:     "HEAD",
+					},
+				},
+			},
+		},
+		"package with alternate repository": {
```

Comment:
I believe this is really testing the logic that you added to the base importer, right? Let's move this testcase into the base importer's tests, as it's not specific to gvt.

Reply:
hm... this tests also tests that I use the `Source` field in `gvt` importer, i.e. it tests this line: https://github.com/michael-go/dep/blob/1cfc27fb29c6369b42d99a51e06e680f1539e819/internal/importers/gvt/importer.go#L116

maybe we just need to added another test in the base importer in addition?

Reply:
Yeah that sounds good, would you please add a test to the base importer as well?